delivery_carrier_package_measure_required/models/stock_picking.py

```diff
@@ -0,0 +1,26 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def _check_required_package_measurement(self, package=None):
+        """Check the required measurement on destination pacakges are set.
+
+        If package is None all result packages on the stock picking are check
+        otherwise only the one.
+        """
+        self.ensure_one()
+        packages = self.move_line_ids.mapped("result_package_id")
+        if package:
+            packages = packages & package
+        for package in packages:
+            package._check_required_dimension()
+
+    def button_validate(self):
+        for pick in self:
```

Comment:
By supporting multi records, your code is simpler: `self._check_required_package_measurement()`